ffi/src/std_impls.rs

```diff
@@ -20,7 +20,7 @@ ffi_type! {unsafe impl<'slice> Transparent for &'slice str[&'slice [u8]] validat
 #[cfg(feature = "non_robust_ref_mut")]
 ffi_type! {unsafe impl<'slice> Transparent for &'slice mut str[&'slice mut [u8]] validated with {|target| core::str::from_utf8(target).is_ok()} }
 ffi_type! {unsafe impl<T> Transparent for core::mem::ManuallyDrop<T>[T] validated with {|_| true} }
-ffi_type! {unsafe impl<T> Transparent for core::ptr::NonNull<T>[*mut T] validated with {|target: &*mut T| target.is_null()} }
+ffi_type! {unsafe impl<T> Transparent for core::ptr::NonNull<T>[*mut T] validated with {|target: &*mut T| !target.is_null()} }
```

Comment:
This, while subtle, should have been caught by tests. The fact that we don't have any, is a little worrying.

Reply:
I concur, but I don't have the bandwidth to test every type. I would be interested to see if OpenAI can help writing these boilerplate tests. I expect it would do an admirable job.

it's not so bad, as long as it's not UB  ...